File: get_current_time.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
获取当前时间的脚本（使用标准库）
"""

from datetime import datetime, timedelta

def get_current_time():
    """获取当前时间"""
    # 获取当前时间
    now = datetime.now()
    
    # 格式化输出
    print("*" * 50)
    print("⏰ 当前时间信息 ⏰")
    print("*" * 50)
    print(f"👉 完整时间: {now.strftime('%Y年%m月%d日 %H:%M:%S')}")
    print(f"👉 ISO格式: {now.isoformat()}")
    print(f"👉 星期: {now.strftime('%A')} ({now.strftime('%w')})")
    print(f"👉 时间戳: {now.timestamp()}")
    print("*" * 50)
    
    return now

def get_weekday_chinese(weekday: int) -> str:
    """将weekday数字转换为中文星期名称"""
    weekday_names = ["周一", "周二", "周三", "周四", "周五", "周六", "周日"]
    return weekday_names[weekday]


def get_time_until_weekend():
    """计算距离周末还有多长时间"""
    now = datetime.now()
    
    # 获取当前是星期几（0=周一，6=周日）
    current_weekday = now.weekday()
    weekday_name = get_weekday_chinese(current_weekday)
    
    print(f"\n📅 今天是: {weekday_name}")
    
    # 计算距离周六（weekday=5）的时间
    if current_weekday < 5:  # 周一到周五
        days_until_saturday = 5 - current_weekday
        
        # 使用更精确的方式计算：直接算到下个周六的0点
        next_saturday = now.replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=days_until_saturday)
        time_diff = next_saturday - now
        total_seconds = time_diff.total_seconds()
        
        # 转换为天时分秒
        days = int(total_seconds // (24 * 3600))
        hours = int((total_seconds % (24 * 3600)) // 3600)
        minutes = int((total_seconds % 3600) // 60)
        seconds = int(total_seconds % 60)
        
        print("\n⏳ 距离周末（周六）还有：")
        print(f"  📆 天数: {days}天")
        print(f"  🕐 小时: {hours}小时")
        print(f"  ⏱️  分钟: {minutes}分钟")
        print(f"  ⏲️  秒数: {seconds}秒")
        print(f"  📊 总分钟数: {total_seconds / 60:.2f}分钟")
        
        # 添加进度条
        total_workdays = 5
        passed_workdays = current_weekday
        progress = passed_workdays / total_workdays
        bar_length = 20
        filled = int(bar_length * progress)
        bar = "█" * filled + "░" * (bar_length - filled)
        print(f"\n  📈 本周进度: [{bar}] {progress * 100:.0f}%")
    else:  # 周六或周日
        print("\n🎉 今天是周末！享受您的休息时间吧！🎉")

def get_greeting() -> str:
    """根据当前时间返回问候语"""
    hour = datetime.now().hour
    if 5 <= hour < 9:
        return "🌅 早上好！新的一天开始了！"
    elif 9 <= hour < 12:
        return "☀️ 上午好！祝您工作顺利！"
    elif 12 <= hour < 14:
        return "🍽️ 中午好！记得吃午饭哦！"
    elif 14 <= hour < 18:
        return "🌤️ 下午好！继续加油！"
    elif 18 <= hour < 22:
        return "🌙 晚上好！辛苦了一天！"
    else:
        return "🌃 夜深了！注意休息哦！"


if __name__ == "__main__":
    # 显示问候语
    print(get_greeting())
    print()
    
    # 获取当前时间
    current_time = get_current_time()
    
    # 计算距离周末的时间
    get_time_until_weekend()
    
    print("\n" + "*" * 50)
    print("感谢使用时间工具！再见！👋")
    print("*" * 50)

File: get_current_time.js
// 获取当前时间的JavaScript脚本

function getCurrentTime() {
    const now = new Date();
    
    console.log("=".repeat(50));
    console.log("当前时间信息");
    console.log("=".repeat(50));
    console.log(`完整时间: ${now.getFullYear()}年${now.getMonth() + 1}月${now.getDate()}日 ${String(now.getHours()).padStart(2, '0')}:${String(now.getMinutes()).padStart(2, '0')}:${String(now.getSeconds()).padStart(2, '0')}`);
    console.log(`ISO格式: ${now.toISOString()}`);
    console.log(`星期: ${['周日', '周一', '周二', '周三', '周四', '周五', '周六'][now.getDay()]}`);
    console.log(`时间戳: ${now.getTime()}`);
    console.log("=".repeat(50));
    
    return now;
}

function getTimeUntilWeekend() {
    const now = new Date();
    const currentDayOfWeek = now.getDay(); // 0=周日, 1=周一, ..., 6=周六
    
    console.log("\n距离周末（周六）还有：");
    
    if (currentDayOfWeek === 6) {
        console.log("今天是周六！享受您的周末吧！");
    } else if (currentDayOfWeek === 0) {
        console.log("今天是周日！享受您的周末吧！");
    } else {
        // 计算到周六的天数
        const daysUntilSaturday = 6 - currentDayOfWeek;
        
        // 计算到今天午夜的时间
        const endOfToday = new Date(now);
        endOfToday.setHours(23, 59, 59, 999);
        const timeUntilMidnight = endOfToday - now;
        
        // 计算总毫秒数
        const totalMilliseconds = timeUntilMidnight + (daysUntilSaturday - 1) * 24 * 60 * 60 * 1000;
        
        // 转换为天时分秒
        const days = Math.floor(totalMilliseconds / (24 * 60 * 60 * 1000));
        const hours = Math.floor((totalMilliseconds % (24 * 60 * 60 * 1000)) / (60 * 60 * 1000));
        const minutes = Math.floor((totalMilliseconds % (60 * 60 * 1000)) / (60 * 1000));
        const seconds = Math.floor((totalMilliseconds % (60 * 1000)) / 1000);
        
        console.log(`天数: ${days} 天`);
        console.log(`小时: ${hours} 小时`);
        console.log(`分钟: ${minutes} 分钟`);
        console.log(`秒数: ${seconds} 秒`);
        console.log(`总分钟数: ${(totalMilliseconds / (60 * 1000)).toFixed(2)} 分钟`);
    }
}

// 执行函数
getCurrentTime();
getTimeUntilWeekend();